docs/sdk/v3/guides/liquidity/04-swap-and-add-liquidity.md

```diff
@@ -0,0 +1,97 @@
+---
+id: swap-and-add
+title: Swapping and Adding Liquidity
+---
+
+## Introduction
+
+This guide will cover how to execute a swap-and-add operation in a single atomic transaction.
+It is based on the [swap-and-add example](https://github.com/Uniswap/examples/tree/main/v3-sdk/swap-and-add-liquidity), found in the Uniswap code examples [repository](https://github.com/Uniswap/examples).
+To run this example, check out the examples's [README](https://github.com/Uniswap/examples/tree/main/v3-sdk/swap-and-add-liquidity) and follow the setup instructions.
+
+:::info
+
+If you need a briefer on the SDK and to learn more about how these guides connect to the examples repository, please visit our [background](./01-background.md) page!
+
+:::
+
+When adding liquidity to a Uniswap v3 pool, you must provide two assets in a particular ratio. In many cases, your contract or the user's wallet hold a different ratio of those two assets. In order to deposit 100% of your assets, you must first swap your assets to the optimal ratio and then add liquidity. 
+
+However, the swap may shift the balance of the pool and thus change the optimal ratio. To avoid that, we can execute this swap-and-add liquidity operation in an atomic fashion, using a router. The inputs to our guide are the **two tokens** that we are pooling for, the **amount** of each token we are pooling for and the Pool **fee**.
+
+The guide will **cover**:
+1. Creating a router instance
+2. Constructing the parameters for the swap-and-add function call
+3. Making the swap-and function call 
+4. Constructing and executing the transaction to swap-and-add liquidity 
+
+At the end of the guide, given the inputs above, we should be able to mint a liquidity position with the press of a button and view the position's id on the UI of the web application. We should also be able to swap-and-add liquidity with the press of a button and see the change reflected in the balance of our tokens.
```

Comment:
Second sentence feels repetitive here, and first doesn't really relate it back to the why here. Maybe something like:

```suggestion
At the end of the guide, we should be able to mint a liquidity position that uses 100% of the input assets with the press of a button and view the position's id on the UI of the web application.
```

Reply:
That's actually not right! We need to first mint the position so we can then swap-and-add liquidity to it. Changed it a bit tho to emphasize on the *100%* piece.